src/main/clojure/conexp/fca/misc.clj

```diff
@@ -235,3 +237,51 @@
                                    0)))))]
     (/ (counter #{} extent)
        (expt 2 (count extent)))))
+
+
+
+(defn concept-probability
+  "Compute the probability of a `concept' in `context' 𝕂 in the following manor.
+  Let pₘ be the relative frequence of attribute m in context. Then the
+  probability of a subset B ⊆ M in 𝕂 is the product of all pₘ for m ∈ B.
+  Then, the probability of a concep is defined by pr(A,B):=pr(B=B'')
+  which is: $\\sum_{k=0}^n {n\\choose k} p_B^k(1-p_B)^{n-k}\\prod_{m\\in
+  M\\setminus B}(1-p_m^k).$"
+  [context concept]
+  (let [n (count (objects context))
+        M (attributes context)
+        B (second concept) ; intent of concept
+        P_M_B (pmap #(/ (count (attribute-derivation context #{%})) n ) (difference M B))
+        p_B (reduce * (pmap #(/ (count (attribute-derivation context #{%})) n) B))
+        k_barrier (/ (Math/log 1E-16) (Math/log (apply max P_M_B)))]
+    (reduce + (pmap (fn [k]
+                       (* (binomial-coefficient n k)
+                          (expt p_B k)
+                          (expt (- 1 p_B) (- n k))
+                          (reduce * (map #(- 1 (expt % k)) P_M_B)))) (range 1 (inc n))))))
+
+(defn faster-concept-probability
+  "In case some of the powers are evaluated in double precision
+  arithmetic we might get loose of some of the last factors. When
+  $p_m^k$ is smallen than the Wilinson Epsilon we might gain no effect
```

Comment:
Docstrings are not meant to contain implementation specific comments; they are meant to users to clarify what this function does (or should do).  Please update this docstring accordingly.